src/VisualStudio/Core/Def/Implementation/Progression/GraphNavigatorExtension.cs

```diff
@@ -73,10 +74,14 @@ public void NavigateTo(GraphObject graphObject)
                         // background thread then the current NewDocumentStateScope is unrelated to
                         // this navigation and it is safe to continue on the UI thread 
                         // asynchronously.
-                        Task.Factory.SafeStartNew(
-                            () => NavigateOnForegroundThread(sourceLocation, symbolId, project, document),
+                        Task.Factory.SafeStartNewFromAsync(
+                            async () =>
+                            {
+                                await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+                                NavigateOnForegroundThread(sourceLocation, symbolId, project, document);
+                            },
                             CancellationToken.None,
-                            ForegroundTaskScheduler);
+                            TaskScheduler.Default);
```

Comment:
you don't use the pattern here you've used elsewhere.  intentional?

Reply:
➡️ It's the same pattern, except bounded by the limitations of starting a new task as opposed to other cases where a task continuation is being scheduled.